woocommerce/woocommerce-ios
# Ensure that an order remains selected after the list filter is changed

Closes: #11750 
Closes: #11752


## Description

- Checks to see if there is a selected order ID, and selects its order
- calling `popToRootViewController` to make sure there are no multiple order details in the hierarchy

## Testing instructions


#11750
- make sure that you have at least 2 orders
- select any order except the first one
- open filters
- select different filter to achieve that the list is not showing previously selected order
- now reset the filter to show all orders
- observe that the previously selected order is selected again (and not the first order in the list)
- make sure to set `splitViewInOrdersTab` to `true`

#11752 
- repro instructions from https://github.com/woocommerce/woocommerce-ios/issues/11752

## Screenshots


![Simulator Screen Recording - iPad Pro (11-inch) (4th generation) - 2024-01-22 at 17 26 32](https://github.com/woocommerce/woocommerce-ios/assets/6242034/7df9d0cd-94ba-454e-95d6-2047a409be99)

---
